Snowflake.Data/Core/Session/SFSessionProperty.cs

```diff
@@ -175,14 +175,14 @@ internal static SFSessionProperties parseConnectionString(String connectionStrin
                                 SFError.INVALID_CONNECTION_STRING,
                                 e.Message);
             }
-            SFSessionProperties properties = new SFSessionProperties();
+            var properties = new SFSessionProperties();
 
-            string[] keys = new string[builder.Keys.Count];
-            string[] values = new string[builder.Values.Count];
+            var keys = new string[builder.Keys.Count];

```

Comment:
I went through the code but I only noticed a refactoring. Were there any behavioural changes?
Do I understand correctly that the code was working well and all you did was refactoring (without behavioural changes) and writing tests?

Reply:
The issue is in this line https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/Session/SFSessionProperty.cs#L207, the condition was value.ToLower() == "WAREHOUSE" so it will always return false. So that was the fixed behavior.